src/wp-includes/class-wp-theme.php

```diff
@@ -1967,14 +1967,21 @@ public function get_block_patterns() {
 	 * Gets block pattern cache.
 	 *
 	 * @since 6.4.0
+	 * @since 6.6.0 Adds support for transient caching.
 	 *
 	 * @return array|false Returns an array of patterns if cache is found, otherwise false.
 	 */
 	private function get_pattern_cache() {
 		if ( ! $this->exists() ) {
 			return false;
 		}
-		$pattern_data = wp_cache_get( 'wp_theme_patterns_' . $this->stylesheet, 'theme_files' );
+
+		if ( wp_using_ext_object_cache() ) {
+			$pattern_data = wp_cache_get( 'wp_theme_patterns_' . $this->stylesheet, 'theme_files' );
+		} else {
+			$pattern_data = get_site_transient( 'theme_files_wp_theme_patterns_' . $this->stylesheet );
```

Comment:
It's probably worth hashing the stylesheet name for the cache key.

It's a super edge case but if the stylesheet is longer than 130ish characters you'll exceed the maximum allowed length for a transient.

Reply:
Done.